src/particles/elements/NonlinearLens.H

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2022 Chad Mitchell, Axel Huebl
+ *
+ * This file is part of ImpactX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_NONLINEARLENS_H
+#define IMPACTX_NONLINEARLENS_H
+
+#include "particles/ImpactXParticleContainer.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuComplex.H>
+
+#include <cmath>
+
+namespace impactx
+{
+    struct NonlinearLens
+    {
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** A thin lens associated with a single short segment of the
+         *  nonlinear magnetic insert described by V. Danilov and
+         *  S. Nagaitsev, PRSTAB 13, 084002 (2010), Sect. V.A.  This
+         *  element appears in MAD-X as type NLLENS.
+         *
+         * @param knll - integrated strength of the nonlinear lens (m)
+         * @param cnll - distance of singularities from the origin (m)
+         *
+         */
+        NonlinearLens( amrex::ParticleReal const knll,
+                   amrex::ParticleReal const cnll )
+        : m_knll(knll), m_cnll(cnll)
+        {
+        }
+
+        /** This is a nonlinear lens functor, so that a variable of this type can be used like a
+         *  nonlinear lens function.
```

Comment:
```suggestion
        /** This is a nonlinear lens functor, so that a variable of this type can
         *  be used like a nonlinear lens function.
```